app/models/period.rb

```diff
@@ -12,4 +12,17 @@
 class Period < ApplicationRecord
   has_many :reports, dependent: :destroy
   has_many :users, through: :reports
+
+  after_create :create_reports_for_bench_users
+
+  private
+      # Create reports for all bench users for a given period
+    def create_reports_for_bench_users
+      User.bench_users.each do |user|
+        Report.create!(
+          user: user,
+          period: self
+        )
+      end
```

Comment:
Since you have an association between `User` and `Report`, you can condense this a little to be something like

```rb
      User.bench_users.each do |user|
        user.reports.create!(period: self)
      end
```
Since you also have an associate between Period and Reports you could do
```rb
      User.bench_users.each do |user|
        self.reports.create!(user: user)
      end
```

I could buy arguments for either one. You might prefer the `user.reports` way because you're looping over _users_ or you might prefer the `self.reports` way because you're in the `Period` class. 

---

Unrelated to this PR, but I noticed you defined the scope `bench_users`

https://github.com/WilsonReese-AppDev/status-reporting/blob/f2b201d1a57436e4f2a3770c825b27bee58a9bfb/app/models/user.rb#L30

With Rails 6, you [automatically get some scopes defined for Enum columns as well as some other things](https://edgeapi.rubyonrails.org/classes/ActiveRecord/Enum.html). So you should automatically have a scope available to you called `on_bench` that is identical to the `bench_users` scope.

Reply:
Amazing. Scopes were still a bit fuzzy to me, so I was looking for an opportunity to make them work. But it is great to know that enums automatically define scopes. I plan to add a couple more enums to my models, so I will make use of that functionality!